server/src/main/java/org/opensearch/index/translog/transfer/FileTransferTracker.java

```diff
@@ -51,32 +56,49 @@ void recordFileTransferStartTime(long uploadStartTime) {
         }
     }
 
-    void recordBytesForFiles(Set<TransferFileSnapshot> toUpload) {
+    void recordBytesForFiles(Set<TranslogCheckpointSnapshot> toUpload) {
         bytesForTlogCkpFileToUpload = new HashMap<>();
         toUpload.forEach(file -> {
-            try {
-                bytesForTlogCkpFileToUpload.put(file.getName(), file.getContentLength());
-            } catch (IOException ignored) {
-                bytesForTlogCkpFileToUpload.put(file.getName(), 0L);
-            }
+            recordFileContentLength(file.getTranslogFileName(), file::getTranslogFileContentLength);
+            recordFileContentLength(file.getCheckpointFileName(), file::getCheckpointFileContentLength);
         });
     }
 
+    private void recordFileContentLength(String fileName, ThrowingSupplier<Long, IOException> contentLengthSupplier) {
+        try {
+            if (!uploaded(fileName)) {
+                bytesForTlogCkpFileToUpload.put(fileName, contentLengthSupplier.get());
+            }
+        } catch (IOException ignored) {
+            bytesForTlogCkpFileToUpload.put(fileName, 0L);
+        }
+    }
+
+    @FunctionalInterface
+    private interface ThrowingSupplier<T, E extends Exception> {
+        T get() throws E;
+    }
```

Comment:
Why do we need this? There are java provided Supplier class already there. Lets remove this please.